web-common/src/features/file-explorer/nav-entry-drag-drop-store.ts

```diff
@@ -75,6 +82,25 @@ export class NavEntryDragDropStore {
     this.dragData.set(this.newDragData);
   }
 
+  public onMouseEnter(dir: string) {
+    this.dropFolders.update((d) => {
+      d.push(dir);
+      return d;
+    });
+    if (!get(this.dragData)) return;
+    setTimeout(
+      () => this.expandDirectory(dir),
+      NavEntryDragDropStore.DROP_HOVER_FOLDER_EXPAND_TIMEOUT,
+    );
+  }
+
+  public onMouseLeave() {
+    this.dropFolders.update((d) => {
+      d.pop();
+      return d;
+    });
+  }
+
   private getOffsets(e: MouseEvent, dragData: NavDragData) {
     const dragItem = document.getElementById(dragData.id);
     if (!dragItem) return;
```

Comment:
**Optimization Issue**: Direct DOM manipulation within Svelte components (document.getElementById) can lead to performance issues and bugs, especially in reactive frameworks where the DOM is supposed to be managed by the framework itself. <br> **Fix**: Use Svelte's binding mechanisms to directly interact with elements, or consider using Svelte actions for more complex interactions that require direct DOM manipulation. <br> **Code Suggestion**: 
 ```
 - const dragItem = document.getElementById(dragData.id);
 + // Consider refactoring to use Svelte's binding or action mechanisms.
 ```

Reply:
We are only getting the rects here so this wont be an issue.